src/main/java/de/tum/in/www1/artemis/service/OrganizationService.java

```diff
@@ -0,0 +1,124 @@
+package de.tum.in.www1.artemis.service;
+
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Organization;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.OrganizationRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+
+/**
+ * Service implementation for managing Organization entities
+ */
+@Service
+public class OrganizationService {
+
+    private final Logger log = LoggerFactory.getLogger(OrganizationService.class);
+
+    private final OrganizationRepository organizationRepository;
+
+    private final UserRepository userRepository;
+
+    private final CourseRepository courseRepository;
+
+    public OrganizationService(OrganizationRepository organizationRepository, UserRepository userRepository, CourseRepository courseRepository) {
+        this.organizationRepository = organizationRepository;
+        this.userRepository = userRepository;
+        this.courseRepository = courseRepository;
+    }
+
+    /**
+     * Performs indexing over all users using the email pattern of the provided organization.
+     * Users matching the pattern will be added to the organization.
+     * Users not matching the pattern will be removed from the organization (if contained).
+     * @param organization the organization used to perform the indexing
+     */
+    public void indexing(final Organization organization) {
```

Comment:
```suggestion
    public void index(final Organization organization) {
```

Reply:
Added in https://github.com/ls1intum/Artemis/pull/3102